hashintel/hash
# Implement `callout` block

## 🌟 What is the purpose of this PR?

 


New block:

![Screenshot 2022-02-15 at 21 28 49](https://user-images.githubusercontent.com/608862/154152363-95ceccf2-59fb-4984-a9f8-36795ffd8f8f.png)

<img width="804" alt="Screenshot 2022-02-15 at 21 54 17" src="https://user-images.githubusercontent.com/608862/154156120-758c3c0a-5049-4418-ae53-1d63ed4c9796.png">

<img width="1409" alt="Screenshot 2022-02-15 at 21 57 50" src="https://user-images.githubusercontent.com/608862/154156406-1ec0446b-1e7e-45ab-aaec-9279a5f0e90e.png">


## ⚠️ Known issues

- `packages/hash/frontend/src/blocks/blocksMeta.tsx: usage of useLocalstorageState() (https://github.com/hashintel/hash/pull/179) breaks syncing with blockPaths.json`

- `packages/hash/shared/src/blockPaths.json`: We cannot add the new block till we’ve merged https://github.com/blockprotocol/blockprotocol/pull/222

- Icon changing is very slow (to do with collab?)

- Icon is changed by flicking through the list of known options. I haven’t added any popovers to keep things simple for now. The design will depend on whether we can support element sticking outside the iframe (in another iframe maybe?)

## 🐾 Next steps

- Merge https://github.com/blockprotocol/blockprotocol/pull/222

- [Add callout to blockPaths.json when two other PRs are merged](https://app.asana.com/0/0/1201818207965387/f) (internal link)

## 🔍 What does this change?

We have a new (simple) callout block.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/0/1201818207965387/f) (internal)

## 🛡 What tests cover this?

- Manual (see instructions below)

## ❓ How to test this?

1.  Checkout the branch and install deps

1.  Open `packages/hash/frontend/src/blocks/blocksMeta.tsx` and make this replacement:

    ```diff
    - const [value, setValue] = useLocalstorageState("blocks-meta", initialValue);
    + const [value, setValue] = useState(initialValue);
    ```

1.  Open packages/hash/shared/src/blockPaths.json and add "https://block.blockprotocol.org/callout": "http://localhost:63212", (e.g. below header)

1.  Run the new block locally

    ```sh
    cd hash/packages/blocks/callout
    yarn build && yarn serve
    ```

1.  Run the app in another terminal (`yarn dev` from monorepo root)

1.  You should be able to create a new Callout block by typing /callout in a blank block.

## 📹 Demo

https://user-images.githubusercontent.com/608862/154151700-4d52d4f2-6689-4cc1-9aa1-78ad3d2c2612.mp4
